aten/src/ATen/native/AdaptiveAveragePooling3d.cpp

```diff
@@ -317,6 +317,12 @@ Tensor adaptive_avg_pool3d_symint(Tensor const& input, SymIntArrayRef output_siz
     // in this case, adaptive pooling is just computing mean over hw
     // dimensions, which can be done more efficiently
     Tensor out = input.mean({-1, -2, -3}, /* keepdim = */ true);
+    if (input.suggest_memory_format() == at::MemoryFormat::ChannelsLast3d) {
+      // assert ndim == 5, since ndim = 4 doesn't give channels_last
+      const auto n = input.sym_size(0);
+      const auto c = input.sym_size(1);
+      out.as_strided__symint({n, c, 1, 1, 1}, {c, 1, c, c, c});
```

Comment:
If I understand correctly, this restride doesn't change the semantic content of the tensor, but mucks around the strides for the size 1 dimensions so it looks channels last 3d.

Reply:
Damn @ezyang I wish I could answer you. I jsut copied and adapted [this code](https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/AdaptiveAveragePooling.cpp#L129-135) the best I could and hoped someone that knows more than me could confirm I didn't do too bad.